Connectors/Revit/Speckle.Connectors.RevitShared/Bindings/RevitSendBinding.cs

```diff
@@ -188,10 +188,48 @@ private void DocChangeHandler(Autodesk.Revit.DB.Events.DocumentChangedEventArgs
       ChangedObjectIds[elementId.ToString()] = 1;
     }
 
-    // TODO: CHECK IF ANY OF THE ABOVE ELEMENTS NEED TO TRIGGER A FILTER REFRESH
+    if (HaveUnitsChanged(e.GetDocument()))
+    {
+      var objectIds = Store.GetSenders().SelectMany(s => s.SendFilter != null ? s.SendFilter.GetObjectIds() : []);
+      _sendConversionCache.EvictObjects(objectIds);
+    }
     _idleManager.SubscribeToIdle(nameof(RevitSendBinding), RunExpirationChecks);
   }
 
+  // Keeps track of doc and current units
+  private readonly Dictionary<string, string> _docUnitCache = new();
```

Comment:
if you switch multiple times in same documents it can cause the return false from `HaveUnitsChanged` by any chance? I don't see anywhere we remove item from this dict

Reply:
oh wait, you override the key